File: src/main/java/astminer/examples/AllJavaFiles.java
package astminer.examples;

import astminer.common.model.LabeledResult;
import astminer.common.model.*;
import astminer.parse.gumtree.java.GumTreeJavaParser;
import astminer.storage.path.Code2VecPathStorage;
import astminer.storage.path.PathBasedStorage;
import astminer.storage.path.PathBasedStorageConfig;
import java.io.FileInputStream;
import java.io.IOException;
import java.nio.file.*;
import java.nio.file.attribute.BasicFileAttributes;

//Retrieve paths from Java files, using a GumTree parser.
public class AllJavaFiles {
    private static final String INPUT_FOLDER = "src/test/resources/gumTreeMethodSplitter";
    private static final String OUTPUT_FOLDER = "out_examples/allJavaFiles_GumTree_java";

    public static void runExample() {
        final PathBasedStorageConfig config = new PathBasedStorageConfig(5, 5, null, null, null);
        final PathBasedStorage pathStorage = new Code2VecPathStorage(OUTPUT_FOLDER, config);

        final Path inputFolder = Paths.get(INPUT_FOLDER);

        FileVisitor<Path> fileVisitor = new SimpleFileVisitor<Path>() {
            @Override
            public FileVisitResult visitFile(Path file, BasicFileAttributes attributes) throws IOException {
                Node fileTree = new GumTreeJavaParser().parseInputStream(new FileInputStream(file.toFile()));

                String filePath = file.toAbsolutePath().toString();
                pathStorage.store(new LabeledResult<>(fileTree, filePath, filePath));

                return FileVisitResult.CONTINUE;
            }
        };

        try {
            Files.walkFileTree(inputFolder, fileVisitor);
            pathStorage.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/astminer/examples/AllJavaFiles.kt
package astminer.examples

import astminer.common.model.LabeledResult
import astminer.parse.antlr.java.JavaFunctionSplitter
import astminer.parse.antlr.java.JavaParser
import astminer.storage.path.Code2VecPathStorage
import astminer.storage.path.PathBasedStorageConfig
import java.io.File

//Retrieve paths from Java files, using a generated parser.
fun allJavaFiles() {
    val inputDir = "src/test/resources/examples/"

    val outputDir = "out_examples/allJavaFilesAntlr"
    val storage = Code2VecPathStorage(outputDir, PathBasedStorageConfig(5, 5))

    File(inputDir).forFilesWithSuffix("11.java") { file ->
        val node = JavaParser().parseInputStream(file.inputStream())
        node.prettyPrint()
        JavaFunctionSplitter().splitIntoFunctions(node, file.path).forEach {
            println(it.name)
            println(it.returnType)
            println(it.enclosingElement?.name)
            it.parameters.forEach { parameter ->
                println("${parameter.name} ${parameter.type}")
            }
        }
        storage.store(LabeledResult(node, file.path, file.path))
    }

    storage.close()
}
